src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1415,6 +1415,16 @@ private bool OpenXmlFileFromPath(string filePath, bool forceManualExecutionMode)
                 var xmlDoc = new XmlDocument();
                 xmlDoc.Load(filePath);
 
+                //save the file before it is migrated to JSON.
+                //if in test mode, don't save the file in backup
+                if (!IsTestMode)
+                {
+                    var fileName = Path.GetFileNameWithoutExtension(filePath) + "_xml";
+                    var extension = Path.GetExtension(filePath);
+                    var savePath = Path.Combine(pathManager.BackupDirectory, fileName + extension);
+                    xmlDoc.Save(savePath);
```

Comment:
the save should be done before the migration.

Reply:
it seems a little odd that if we fail to save a backup we won't open the file. Is that the intention? What if the user doesn't have write access to where they open the file?

Reply:
@mjkkirschner The backup file is default to save to `pathManager.BackupDirectory` which is userData->currentUser->Roaming, this path usually current user always have write access. `savePath` here is different than `filePath`, if we want to have more clue, we can wrap it with try catch and error log just so open does not fail and console will show error. But again, the default backup path seems to good and there is not a way for user to overwrite it yet I think.

Reply:
what happens in the current backup system? if there is only read access, we throw the exception which is not captured on the Dynamo model.

Reply:
Now, I check for folder permission before I write to the folder. But this is only for XML. I haven' checked this for the current backup system (since Aaron has other PR for checking that).